coucharchive

```diff
@@ -307,10 +307,22 @@ class ReplicationControl(object):
         # Choose the value to achieve replication in ideal_duration:
         ideal_number = self._ideal_number_of_replications_for_ideal_duration()
 
-        # ... but if there were errors, take 0.9 × the lowest error value
-        if self._last_errors:
-            min_number_with_error = min(e[1] for e in self._last_errors)
-            ideal_number = int(0.9 * min_number_with_error)
+        # ... but if there were errors, compute an age-weighted average
+        # between past errors and ideal_number.
```

Comment:
I propose:
```diff
--- a/coucharchive
+++ b/coucharchive
@@ -308,12 +308,26 @@ class ReplicationControl(object):
         ideal_number = self._ideal_number_of_replications_for_ideal_duration()
 
         # ... but if there were errors, compute an age-weighted average
-        # between past errors and ideal_number.
+        # between past errors and ideal_number:
+        #
+        #                      ideal
+        #                      target
+        #       last errors,
+        #       age-weighted   +
+        #                      |
+        #                  +   |          new target = weighted average
+        #                  |   |
+        #           +      |   |
+        #    +    + +      +   +
+        #  ----------------------> t
+        #        5 minutes
         now = time.time()
         weights = []
         weighted_values = []
         target = [(now, ideal_number)]
-        # Multiples last errors value by 0.9 so we stay just below
+        # Lower last errors values a bit (× 0.9) so we stay just below the
+        # "error zone":
         last_errors_lowered = [(i[0], i[1] * 0.9) for i in self._last_errors]
         for i in last_errors_lowered + target:
             age = now - i[0]
```

Reply:
Thanks for this improvement, I use the diff!